scripts/assign.py

```
@@ -207,70 +207,146 @@ def main():
+                                .get("nodes", [])
+                            )
+
+                            for item in timeline_items:
+                                source = item.get("source", {})
+                                if source and "state" in source and source["state"] == "OPEN":
+                                    pr_number = source.get("number")
+                                    print(f"Found open PR #{pr_number} linked to issue #{current_issue_number}")
+                                    has_pr = True
+                                    break
+                    except Exception as e:
+                        print(f"Error checking for linked PRs via GraphQL: {str(e)}")
```

Ruff (0.8.2)

Use explicit conversion flag

Replace with conversion flag

(RUF010)